Improve spec file

kgaillot opened the pull request.

jnpkrn commented on pacemaker.spec.in:
```diff
@@ -1,77 +1,113 @@
 %global pcmk_docdir %{_docdir}/%{name}
 
-%global specversion 1
+## Upstream pacemaker version, and its package version 
```
adds trailing whitespace

jnpkrn commented on pacemaker.spec.in:
```diff
@@ -1,77 +1,113 @@
+%global specversion 1
+
+## Upstream commit (or git tag, such as "Pacemaker-${pcmkversion}"
+## for an official release) to use for this package
```
`${pcmkversion}` will not work per se and `%{pcmkversion}` should not be used as macros are expanded whether in comment or not.  Would suggest `s/%%/%/` (poor eyes) or keep referring just to `"pcmkversion" macro` (which is what i used originally).

jnpkrn replied:
(the note about the eyes was my feelign when writing that sed expression in standard proportial font, monospaced font should be default, really)

jnpkrn replied:
Anyway, packagers should understand even the dollar version.

jnpkrn replied:
One more thing, you perhaps mean to expand `pcmkversion` in-situ, while I definitely preferred to use the symbolic value referenced by that very macro (so that the macro is the only authoritative holder of that value, leaving possibility the two would differ behind.

jnpkrn commented on pacemaker.spec.in:
```diff
@@ -1,77 +1,113 @@
+## Whether this is a development branch
 %global post_release %([ %{commit} = Pacemaker-%{shortcommit} ]; echo $?)
+
+## Github organization that distributes source
```
Would use something like:
```
This spec tight matches GitHub URL scheme as this is where the source is hosted;
this allows to change "owner" part of the scheme for a forks' convenience.
```

jnpkrn commented on pacemaker.spec.in:
```diff
@@ -1,77 +1,113 @@
+## so that the -devel package is multilib-compliant
 %global __os_install_post %(echo '%{__os_install_post}' | sed -e 's!/usr/lib[^[:space:]]*/brp-python-bytecompile[[:space:]].*$!!g')
 
+## Heuristic used to infer brokenness of some tools
```
`Heuristic assuming that a bleeding-edge distro will have some tools broken`?

jnpkrn replied:
Or is it meant for the list growing in the future?  We should rename the variable then.

kgaillot commented:
Updated per review, rebased to include recent changes, and added a commit to remove %ifarch.

What is your opinion of 9d76684? I think it will help when reviewing diffs, but I'm not sure whether it's more or less understandable.

jnpkrn commented:
Re 9d76684: I was neutral in previous go, now when there's init-system overhaul in the works (mostly affecting `configure.ac`), I downright like it.

jnpkrn commented on pacemaker.spec.in:
```diff
@@ -1,77 +1,114 @@
+## GitHub entity that distributes source (for ease of using a fork)
+%global github_owner ClusterLabs
+
+## Upstream pacemaker version, and its package version
```
From my experience, it would deserve, akin to previous explanation, `(leveraged in automation)` (meaning "make (s)rpm", indeed).

jnpkrn commented:
Regarding that particular `ifarch` drop, that's fine to declare clearly we cannot ever subsume all possible architectures from the past or present, not speaking about new ones to come (such as [RISC-V](https://fedoraproject.org/wiki/Architectures/RISC-V)), so RPM packaging fine tuning is left as an exercise to the enthusiasts.
